implementations/scripts/utils.py

```diff
@@ -0,0 +1,63 @@
+import datetime
+import json
+
+
+def str_to_dt_other(date):
+    """
+    :param date: date (a string with format "%Y-%m-%d")
+    Converts date to a datetime object if date has the above format
+    """
+    datetimeobj = datetime.datetime.strptime(date, "%Y-%m-%d")
+    return datetimeobj
+
+
+def str_to_dt_data(date):
+    """
+    :param date: converts date (str) to a datetime object
+    Note: the string format for the date in the json file is either:
+     - %a %b %d %H:%M:%S %Y %z --> for commits
+     - %Y-%m-%dT%H:%M:%SZ      --> for issues and pull requests
+    """
+    try:
+        datetimestr = datetime.datetime.strptime(
+            date, "%a %b %d %H:%M:%S %Y %z").strftime("%Y-%m-%d")
+
+    except ValueError as ve:
+        datetimestr = datetime.datetime.strptime(
+                date, "%Y-%m-%dT%H:%M:%SZ").strftime("%Y-%m-%d")
+
+    finally:
+        datetimeobj = datetime.datetime.strptime(datetimestr, "%Y-%m-%d")
+        return datetimeobj
+
+
+def get_date(df, option="since"):
```

Comment:
I'm not sure this is really needed, but we can discuss later, during the next iteration.

Reply:
The reason I added this was for the `compute_timeseries` method: I needed a way to make sure that the dataframe returned was between a definite range. If no `since` or `until` are provided, I made it so that the earliest event in the JSON data would become `since` and the latest would be `until`.

Reply:
Yes, I understood that, but maybe the timeseries can be constructed without this. However, leave it this way, if you want. We can enter in details during the next iteration.